pkg/firefly/firefly.go

```diff
@@ -0,0 +1,173 @@
+// Copyright © 2022 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package firefly
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/hyperledger/firefly-common/pkg/config"
+	"github.com/hyperledger/firefly-common/pkg/i18n"
+	"github.com/hyperledger/firefly-common/pkg/log"
+	"github.com/hyperledger/firefly/internal/apiserver"
+	"github.com/hyperledger/firefly/internal/coreconfig"
+	"github.com/hyperledger/firefly/internal/namespace"
+	"github.com/sirupsen/logrus"
+)
+
+const configSuffix = "core"
+
+var sigs = make(chan os.Signal, 1)
+
+var _utManager namespace.Manager
+
+func ResetConfig() {
+	coreconfig.Reset()
+	apiserver.InitConfig()
+}
+
+func ShowConfig(cfgFile string) string {
+	// Initialize config of all plugins
+	ResetConfig()
+	GetRootManager()
+	_ = config.ReadConfig(configSuffix, cfgFile)
+
+	// Print it all out
+	result := fmt.Sprintf("%-64s %v\n", "Key", "Value")
+	result = fmt.Sprintf("%s-----------------------------------------------------------------------------------\n", result)
+	for _, k := range config.GetKnownKeys() {
+		result = fmt.Sprintf("%s%-64s %v\n", result, k, config.Get(config.RootKey(k)))
+	}
+	return result
+}
+
+func GetRootManager() namespace.Manager {
+	if _utManager != nil {
+		return _utManager
+	}
+	return namespace.NewNamespaceManager(true)
+}
+
+func Run(cfgFile string) error {
+
+	// Read the configuration
+	ResetConfig()
+	err := config.ReadConfig(configSuffix, cfgFile)
```

Comment:
Sorry for having a hazy memory of the interface here on Viper, but is this actually the file, or is this a filename prefix that is used for searching?

I ask because firstly to confirm there's no change in behavior here, but hen also I think this has been a point of confusion in the code before - so it might be good to be explicit in the comments as an additional improvement here.